com.unity.render-pipelines.universal/Editor/AssetPostProcessors/ForwardToUniversalRendererPostprocessor.cs

```diff
@@ -3,57 +3,91 @@
 using UnityEngine.Rendering;
 using UnityEngine.Rendering.Universal;
 
-class ForwardToUniversalRendererPostprocessor : AssetPostprocessor
+namespace UnityEditor.Rendering.Universal
 {
-    static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
+    class ForwardToUniversalRendererPostprocessor : AssetPostprocessor
     {
-        //Gets the UniversalRendererData.cs file
-        string stdRendererScriptFilePath = AssetDatabase.GUIDToAssetPath("f971995892640ec4f807ef396269e91e");
-        var stdRendererScriptObj = AssetDatabase.LoadAssetAtPath(stdRendererScriptFilePath, typeof(Object));
+        static int editedAssetsCount = 0;
 
-        //Gets the ForwardRendererData.cs file
-        string fwdRendererScriptFilePath = AssetDatabase.GUIDToAssetPath("de640fe3d0db1804a85f9fc8f5cadab6");
-        var fwdRendererScriptObj = AssetDatabase.LoadAssetAtPath(fwdRendererScriptFilePath, typeof(Object));
-
-        int noOfFwdRendererData = 0;
-
-        //Gets all the Renderer Assets in project
-        Object[] allRenderers = Resources.FindObjectsOfTypeAll(typeof(ScriptableRendererData));
-        for (int i = 0; i < allRenderers.Length; i++)
+        static void UpgradeAsset(Object rendererData, string rendererDataPath, Object fwdRendererScriptObj, Object stdRendererScriptObj)
         {
-            SerializedObject so = new SerializedObject(allRenderers[i]);
+            if (rendererData == null) return;
+            SerializedObject so = new SerializedObject(rendererData);
+
+            //Double check to see if it's using ForwardRendererData
             SerializedProperty scriptProperty = so.FindProperty("m_Script");
-            so.Update();
-            //check if the script is using ForwardRendererData
-            if (scriptProperty.objectReferenceValue.Equals(fwdRendererScriptObj))
+            if (scriptProperty.objectReferenceValue == fwdRendererScriptObj)
             {
-                //change the script to use UniversalRendererData
+                //Change the script to use UniversalRendererData
+                so.Update();
                 scriptProperty.objectReferenceValue = stdRendererScriptObj;
                 so.ApplyModifiedProperties();
-                noOfFwdRendererData++;
+
+                //Save and re-import asset
+                //This prevents the "Importer(NativeFormatImporter) generated inconsistent result" warning
+                AssetDatabase.SaveAssets();
```

Comment:
Minor: Do we need to call AssetDatabase.SaveAssets() for each asset? AssetDatabase.SaveAssets() is expensive operation.

Reply:
I tested the big UniversalGraphicsTest on my Mac, doesn't look like it causes any importing time differences.

without SaveAssets():
153.282+1.569 seconds = 154.851
150.601+1.531 seconds = 152.132
with SaveAssets()
148.142+1.230 seconds = 149.372
151.260+1.629 seconds = 152.889

But anyway I tested both test cases without SaveAssets() and the upgrade still works. So I've removed SaveAssets() in latest commit. Will merge the PR when Yamato tests are good as well.